Improve documentation (#13)
anbox-cloud/cloud-gaming-demo: README.md

Inside the web UI, the user can choose from the list of available games and start playing either of them.

![Cloud Gaming Demo UI](docs/screenshot.png)

> ## Install

Reviewer: ```suggestion
## How to install
```

At the end, the installation script prints the URL where you can access the demo. Happy gaming!

## Uninstall

In case that you don't want to use the cloud gaming demo any longer you can simply uninstall it from
your Anbox Cloud Appliance by running the following commands

    cd cloud-gaming-demo
    sudo ./uninstall.sh

Afterwards all configuration and the cloud gaming demo itself is removed from the machine.

## Used games

The demo uses the following games:

 * [BombSquad](http://www.bombsquadgame.com/)

>  * [Mindustry](https://github.com/Anuken/Mindustry)

Reviewer 2: As we don't use the game `Minetest`, could you
1. drop the `minetest` from the `GAMES` array in the `scripts/install.sh`.
2. update the screenshot to honor this change as well.

Author: 1. is already done in https://github.com/anbox-cloud/cloud-gaming-demo/pull/12  and 2. I would like to keep as is

Reviewer 2: Make sense to me.

At the end, the installation script prints the URL where you can access the demo. Happy gaming!

> ## Uninstall
> 
> In case that you don't want to use the cloud gaming demo any longer you can simply uninstall it from
> your Anbox Cloud Appliance by running the following commands
> 
>     cd cloud-gaming-demo
>     sudo ./uninstall.sh
> 
> Afterwards all configuration and the cloud gaming demo itself is removed from the machine.

Reviewer: ```suggestion
## How to uninstall

If you don't want to use the cloud gaming demo any longer, simply uninstall it from
your Anbox Cloud Appliance by running the following commands:

    cd cloud-gaming-demo
    sudo ./uninstall.sh

This script removes the cloud gaming demo and all its configuration from the machine.
```

At the end, the installation script prints the URL where you can access the demo. Happy gaming!

## Uninstall

In case that you don't want to use the cloud gaming demo any longer you can simply uninstall it from
your Anbox Cloud Appliance by running the following commands

    cd cloud-gaming-demo
    sudo ./uninstall.sh

Afterwards all configuration and the cloud gaming demo itself is removed from the machine.

## Secure access via HTTP basic auth

> By default the demo is exposed without access control. However you can easily add HTTP basic auth by changing
> the [traefik](https://traefik.io/) configuration at `/var/snap/anbox-cloud-appliance/common/traefik/conf/cloud-gaming-demo.yaml`
> by adding a HTTP basic auth middleware definition (see [here](https://doc.traefik.io/traefik/v2.0/middlewares/basicauth/)
> for more details).

Reviewer: ```suggestion
By default, the demo is exposed without access control. However, you can easily add basic HTTP authentication by adding a middleware definition (see [BasicAuth](https://doc.traefik.io/traefik/v2.0/middlewares/basicauth/) for more details) to the [traefik](https://traefik.io/) configuration at `/var/snap/anbox-cloud-appliance/common/traefik/conf/cloud-gaming-demo.yaml`.
```

> First we have to use the `htpasswd` tool to generate a user/password combination

Reviewer: ```suggestion
First, use the `htpasswd` tool to generate a user/password combination:
```

    apt install -y apache2-utils
    httpasswd -n <your user name>

> On the prompt enter your desired password. The printed user/hashed password combination can now be
> inserted into the traefik configuration. For that open `/var/snap/anbox-cloud-appliance/common/traefik/conf/cloud-gaming-demo.yaml`
> in an editor and change the configuration to look like:

Reviewer: ```suggestion
Enter your desired password when prompted. Then insert the printed user/hashed password combination into the traefik configuration by opening `/var/snap/anbox-cloud-appliance/common/traefik/conf/cloud-gaming-demo.yaml`
in an editor and changing the configuration to look like:
```

    http:
    routers:
        ...
        middlewares: ["ratelimiter", "strip-demo-prefix", "demo-auth"]
    middlewares:
        ...
        demo-auth:
            basicAuth:
                users:
                - "<user name>:<hashed password>"
    ...

> Afterwards every user will be asked for username and password when accessing the demo site.

Reviewer: ```suggestion
Afterwards, every user will be asked for a user name and password when accessing the demo site.
```

> ## Build

Reviewer: ```suggestion
## How to build a snap
```

> If you want to build the [snap package](https://snapcraft.io) including the demo UI and service yourself
> you first have to install the [snapcraft](https://snapcraft.io) build tool. Please find further details
> [here](https://snapcraft.io/docs/snapcraft-overview)

Reviewer: ```suggestion
If you want to build the [snap package](https://snapcraft.io) including the demo UI and service yourself,
install the [snapcraft](https://snapcraft.io) build tool. See the [Snapcraft documentation](https://snapcraft.io/docs/snapcraft-overview) for instructions.
```

> Once you have snapcraft installed all you have to do is running the following command:
> 
>     snapcraft
> 
> This will start the build process and procude a `.snap` package at the end which you can install with

Reviewer: ```suggestion
When you have snapcraft installed, run the following command in the root directory of the repository to build the snap:

    snapcraft

This will start the build process and produce a `.snap` package that you can install with the following command:
```

Reviewer: I guess you run it in the root folder?

    snap install --dangerous cloud-gaming-demo*.snap

## Used games

> The demo uses the following games:

Reviewer: Maybe make that a bit weaker? In case we forget to update it in the future. ;)
```suggestion
The demo currently uses the following games:
```
